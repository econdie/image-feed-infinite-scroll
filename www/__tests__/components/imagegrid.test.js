import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import ImageGrid from "../../components/common/imagegrid";
import { Card, CardImg, CardText, CardDeck, CardBody, Alert } from "reactstrap";

const images = [
  {
    imageUrl: "https://picsum.photos/id/80/200",
    caption: "Exercitationem eum impedit et et quam debitis."
  },
  {
    imageUrl: "https://picsum.photos/id/81/200",
    caption: "Id autem rerum et."
  },
  {
    imageUrl: "https://picsum.photos/id/82/200",
    caption:
      "Id doloribus ipsam eum aliquam excepturi repellendus eum officiis odio."
  }
];

describe("ImageGrid", () => {
  it("renders expected number of components", () => {
    const grid = shallow(<ImageGrid images={images} />);
    expect(grid.find(CardDeck)).toHaveLength(1);
    expect(grid.find(Card)).toHaveLength(3);
    expect(grid.find(CardImg)).toHaveLength(3);
    expect(grid.find(CardText)).toHaveLength(3);
    expect(grid.find(Alert).find("#finishedLoading")).toHaveLength(0);
    expect(
      grid.find(CardImg).find({ src: "https://picsum.photos/id/80/200" })
    ).toHaveLength(1);
  });

  it("renders no more images message", () => {
    const grid = shallow(<ImageGrid images={images} moreItems={false} />);
    expect(grid.find(Alert).find("#finishedLoading")).toHaveLength(1);
  });

  it("Snapshot matches", () => {
    const component = renderer.create(<ImageGrid images={images} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
