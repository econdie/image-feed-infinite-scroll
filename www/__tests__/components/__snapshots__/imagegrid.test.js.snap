// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGrid Snapshot matches 1`] = `
<div
  className="card-deck"
>
  <div
    className="container"
  >
    <div>
      <div
        className="row"
      >
        <div
          className="my-2 col-sm-12 col-md-6 col-lg-4 col-xl-4"
        >
          <div
            className="card"
          >
            <img
              className="card-img-top"
              src="https://picsum.photos/id/80/200"
            />
            <div
              className="card-body"
            >
              <p
                className="card-text"
              >
                Exercitationem eum impedit et et quam debitis.
              </p>
            </div>
          </div>
        </div>
        <div
          className="my-2 col-sm-12 col-md-6 col-lg-4 col-xl-4"
        >
          <div
            className="card"
          >
            <img
              className="card-img-top"
              src="https://picsum.photos/id/81/200"
            />
            <div
              className="card-body"
            >
              <p
                className="card-text"
              >
                Id autem rerum et.
              </p>
            </div>
          </div>
        </div>
        <div
          className="my-2 col-sm-12 col-md-6 col-lg-4 col-xl-4"
        >
          <div
            className="card"
          >
            <img
              className="card-img-top"
              src="https://picsum.photos/id/82/200"
            />
            <div
              className="card-body"
            >
              <p
                className="card-text"
              >
                Id doloribus ipsam eum aliquam excepturi repellendus eum officiis odio.
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="text-center my-4"
      >
        <div
          className="spinner-border text-primary"
          role="status"
        >
          <span
            className="sr-only"
          >
            Loading...
          </span>
        </div>
         Loading..
      </div>
    </div>
  </div>
</div>
`;
